### ISSUE TITLE
Support Debug Adapter Protocol

### ISSUE THREAD

**Issue by pmetzger** [realgud/realgud#231]
Microsoft has created a new protocol that is a sort of debugger analog to the Language Server Protocol. See: https://microsoft.github.io/debug-adapter-protocol/

It might make sense to have realgud support it.

**Comment by rocky** [comment]
Yes, it seems like a good idea. I am sure it would happen sooner if you took the lead on this and started working on it.

**Comment by pmetzger** [comment]
I wanted to put a pin on it in by opening a ticket so that it wouldn't be forgotten. Whether I eventually have time to work on it is hard to say, certainly no one should wait for me to get started.

**Comment by yyoncho** [comment]
@pmetzger FYI there is an implementation of DAP client at https://github.com/yyoncho/dap-mode/ which could be used on top of lsp-mode since a lot of the so-called language specific adapter relly on the corresponding language server(e. g. the Java Debugger depends on JDT LS for loading the classpath).
